packages/grid/_modules_/grid/components/toolbar/GridToolbarExport.tsx

```diff
@@ -7,28 +7,44 @@ import MenuItem from '@material-ui/core/MenuItem';
 import { isHideMenuKey, isTabKey } from '../../utils/keyboardUtils';
 import { GridApiContext } from '../GridApiContext';
 import { GridMenu } from '../menu/GridMenu';
-import { GridExportOption } from '../../models';
+import { GridExportCsvOptions } from '../../models/gridExport';
 
-export const GridToolbarExport = React.forwardRef<HTMLButtonElement, ButtonProps>(
+interface GridExportFormatCsv {
+  format: 'csv';
+  formatOptions?: GridExportCsvOptions;
+}
+
+type GridExportFormatOption = GridExportFormatCsv;
```

Comment:
Why do we need these 2 types, why not only 1?
Not sure everybody would agree but I would export the type used in the props, so users can easily type the component props.

Reply:
because as soon as we have excel, it needs:

```suggestion
type GridExportFormatOption = GridExportFormatCsv | GridExportFormatExcel;
```

Reply:
If you export the button to recompose your toolbar and you allow to pass props in the button, you might want to type those props and what's inside them

Reply:
> If you export the button to recompose your toolbar and you allow to pass props in the button, you might want to type those props and what's inside them

True, isn't the `GridToolbarExportProps` type already public for this use case?

You can use https://codesandbox.io/s/material-demo-forked-o7cee?file=/demo.tsx if you were thinking of something else, it uses the last commit of this PR.